### ISSUE TITLE
Add setting to transform from right join to left join and vice versa

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10650]
`LEFT JOIN` and `RIGHT JOIN` are almost the same thing. The only difference is which table is the one that is joined optionally, so these two statements are equivalent:

```sql
SELECT a.*, b.* FROM a LEFT JOIN b ON p;
SELECT a.*, b.* FROM b RIGHT JOIN a ON p;
```

Notice that explicit projections are made. If projecting `*`, the columns appear in the order of tables in the join graph, so the two join types are not *exactly* equivalent.

A new setting could allow for transforming the two join types into each other, which is mostly helpful on https://www.jooq.org/translate, in these two use-cases:

- [ ] There are `RIGHT JOIN` expressions that people want to clean up for stylistic reasons
- [ ] There are `RIGHT JOIN` expressions that are not supported on an RDBMS that people migrate to

This might be useful to further transform `RIGHT JOIN` generated from the `Settings.transformTableListsToAnsiJoin` transformation #10697, if a target dialect doesn't support `RIGHT JOIN`

**Comment by lukaseder** [comment]
Apache Ignite could profit from this: https://github.com/jOOQ/jOOQ/issues/10551

**Comment by lukaseder** [comment]
This is also a reasonable pattern transformation, see `transformPatternsNormaliseRightJoinToLeftJoin` in https://github.com/jOOQ/jOOQ/issues/13593, where the point is normalising SQL to simplify other patterns. The logic could be made reusable, because it's also useful when `RIGHT JOIN` isn't supported

**Comment by lukaseder** [comment]
Giving this more thought, with Redshift being the last jOOQ dialect that finally supports `RIGHT JOIN` (https://github.com/jOOQ/jOOQ/issues/14249), there's no need for this emulation anymore. The transformation will be made available soon with #13593
